pkg/diagnostics/utils/baggage.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright 2025 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"strings"
+
+	diagConsts "github.com/dapr/dapr/pkg/diagnostics/consts"
+	otelbaggage "go.opentelemetry.io/otel/baggage"
+)
+
+// ProcessBaggageValues handles baggage header validation, property parsing, and member creation.
+func ProcessBaggageValues(baggageValues []string) ([]string, []otelbaggage.Member) {
```

Comment:
Why are we not using otel for this?

https://pkg.go.dev/go.opentelemetry.io/otel/baggage

Reply:
otel diverged from w3 sometimes with limits or propagates and empty baggage, so I was trying to do a bit of both. Ended up switching to align fully with otel